src/components/bottomMenu.tsx

```diff
@@ -0,0 +1,146 @@
+
+export const BottomMenu = forwardRef(({ menu }: { menu: MenuComponent }, ref) => {
+
+    // hooks
+    const sheetRef = useRef<BottomSheet>(null)
+    const snapPoints = useMemo(() => ['0%', '80%'], [])
+
+    const handleSheetChange = useCallback(index => {
+        console.log('handleSheetChange', index)
+    }, [])
+
+    useImperativeHandle(ref, () => ({
+        handleOpen() {
+            sheetRef.current?.snapTo(1)
+        }
+    }))
+
+    const handleClose = useCallback(() => {
+        sheetRef.current?.close()
+    }, [])
+
+    const handleShare = useCallback(() => {
+        sheetRef.current?.close()
+        Share.share({
+            message: `https://app.loop.page/b/${menu.block_id}`
+        })
+    }, [])
+
+    const renderItem = useCallback(
+        (title, onClick, subtitle?, icon?) => (
+            <TouchableRipple onPress={onClick}>
+                <>
+                    <View key={title} style={styles.itemContainer}>
+                        <Text style={styles.itemTitle}>{title}</Text>
+                        <View style={styles.itemRightView}>
+                            {icon != null ? <MaterialCommunityIcons style={styles.icon} name={icon} color={colors.white} size={20} /> : null}
+                            {subtitle != null ? <Text style={styles.itemSubtitle}>{subtitle}</Text> : null}
+                        </View>
+                    </View>
+                    <Divider style={styles.separator} />
+                </>
+            </TouchableRipple>
+        ), []
+    )
+
+    const backgroundComponent = ({ style }: BottomSheetBackgroundProps) => {
+        return <View style={[style, styles.bottomSheet]} />
+    }
+
+    const backdropComponent = (props: BottomSheetBackdropProps) => {
+        return <BottomSheetBackdrop {...props} enableTouchThrough={true} />
+    }
+
+    return (
+        <Portal>
+            <BottomSheet
+                ref={sheetRef}
+                index={-1}
+                backgroundComponent={backgroundComponent}
+                backdropComponent={backdropComponent}
+                snapPoints={snapPoints}
+                onChange={handleSheetChange}>
+                <BottomSheetScrollView>
+                    {menu.star_button &&
+                        renderItem(
+                            menu.star_button.starred ? 'Unstar' : 'Star',
+                            handleClose,
+                            menu.star_button.count,
+                            menu.star_button.starred ? 'star' : 'star-outline')
+                    }
+                    {menu.notifications_enabled &&
+                        renderItem(
+                            menu.notifications_enabled ? 'Disable Notification' : 'Enable Notification',
+                            handleClose,
+                            null,
+                            menu.notifications_enabled ? 'bell' : 'bell-off')
+                    }
+                    {menu.delete &&
```

Comment:
Could you remove the delete button b/c there's no ticket & implementation is broken

Reply:
Removed